### PR TITLE
fix(filters): Wrong options in event status filter for tracker and events

### PR DESCRIPTION
**Pull request by Balcan**
This PR has changes from this two related jira issues:
[jira issue 1](https://jira.dhis2.org/browse/ANDROAPP-2545)
[jira issue 2](https://jira.dhis2.org/browse/ANDROAPP-2549)

### CODE REVIEW

File: app/src/main/java/org/dhis2/usescases/main/program/HomeRepositoryImpl.kt [hunk 1]
```diff
@@ -10,360 +10,383 @@ import org.hisp.dhis.android.core.dataset.DataSet
...
+import org.hisp.dhis.android.core.event.EventStatus
 import org.hisp.dhis.android.core.organisationunit.OrganisationUnit
 import org.hisp.dhis.android.core.period.DatePeriod
 import org.hisp.dhis.android.core.program.ProgramCollectionRepository
 import org.hisp.dhis.android.core.program.ProgramType.WITHOUT_REGISTRATION
 import org.hisp.dhis.android.core.program.ProgramType.WITH_REGISTRATION
+import java.util.*
+import kotlin.collections.ArrayList
 
 internal class HomeRepositoryImpl(private val d2: D2, private val eventLabel: String) :
```
Comment (ferdyrod) [inline app/src/main/java/org/dhis2/usescases/main/program/HomeRepositoryImpl.kt]: KtlintFormat was not execute in this file. imports are not in lexicographical order.

**Comment by Balcan** [comment]
There are 2 jira issues related to this PR. Please check both of them.

File: app/src/main/java/org/dhis2/usescases/main/program/ProgramViewModel.kt [hunk 2]
```diff
@@ -35,65 +35,95 @@ abstract class ProgramViewModel : BaseObservable(), Serializable {
...
-            count: Int,
-            type: String?,
-            typeName: String,
-            programType: String,
-            description: String?,
-            onlyEnrollOnce: Boolean,
-            accessDataWrite: Boolean
+        /*fun create(
+                uid: String,
+                displayName: String,
```
Comment (ferdyrod) [inline app/src/main/java/org/dhis2/usescases/main/program/ProgramViewModel.kt]: commented code.

**Comment by ferdyrod** [comment]
tested on devices and it's working as expected.

File: app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java [hunk 3]
```diff
@@ -592,6 +611,30 @@ else if (d2.trackedEntityModule().trackedEntityAttributes().byUid().eq(tetAttrib
...
+                    try {
+                        d2.eventModule().events().uid(event.uid()).setStatus(EventStatus.OVERDUE);
+                    } catch (D2Error d2Error) {
+                        d2Error.printStackTrace();
+                    }
+                }
+
+                boolean hasEventWithStatus = !d2.eventModule().events().byTrackedEntityInstanceUids(Collections.singletonList(tei.uid())).byStatus().in(eventStatuses).blockingIsEmpty();
+                if (!hasEventWithStatus)
+                    iterator.remove();
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java]: I don't think this pass the code style from lint :dancer:

Comment (Balcan) [reply app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java]: It's java :P

**Pull request merged**